compiler/rustc_codegen_cranelift/src/constant.rs

```diff
@@ -513,7 +513,7 @@ pub(crate) fn mir_operand_get_const_val<'tcx>(
                         {
                             return None;
                         }
-                        StatementKind::CopyNonOverlapping(_) => {
+                        StatementKind::Assume(_) | StatementKind::CopyNonOverlapping(_) => {
```

Comment:
Assume makes more sense for the branch below as it never writes. It doesn't really matter much though. This is basically a big hack to evaluate things like `IMM8 as u32` to constants. Using `const {}` in stdarch should allow removing this code. Not sure if `const {}` can refer to const generics though.